services/ymax-planner/src/sequence-manager.ts

```diff
@@ -0,0 +1,108 @@
+import { Fail } from '@endo/errors';
+import type { CosmosRestClient } from './cosmos-rest-client.ts';
+
+type AccountInfo = {
+  '@type': string;
+  address: string;
+  pub_key?: {
+    '@type': string;
+    key: string;
+  };
+  account_number: string;
+  sequence: string;
+};
+
+type AccountResponse = {
+  account: AccountInfo;
+};
+
+type SequenceManagerConfig = {
+  address: string;
+  chainKey: string;
+};
+
+type SequenceManagerPowers = {
+  cosmosRest: CosmosRestClient;
+  log?: (...args: unknown[]) => void;
+};
+
+export class SequenceManager {
+  private sequence: number = 0;
+  private accountNumber: number = 0;
+  private readonly address: string;
+
+  private readonly cosmosRest: CosmosRestClient;
+  private readonly chainName: string;
+
+  private initialized = false;
+  private readonly log: (...args: unknown[]) => void;
+
+  constructor(io: SequenceManagerPowers, config: SequenceManagerConfig) {
+    this.address = config.address;
+    this.chainName = config.chainKey;
+    this.cosmosRest = io.cosmosRest;
+    this.log = io.log ?? (() => {});
```

Comment:
IIUC, `this.log` is not encapsulated and it's mutable. So I'd have to check all the code in the process to know whether it gets clobbered. Likewise all the other properties and methods, I suppose.